### PR TITLE
Feat/decouple buttons

### PR DESCRIPTION
**Pull request by azakordonets**
- Added eslint config with rules
- Fixed errors from eslint output in some files

Note: i had to put max length of one line of code to 170 because of one big regex that i couldn't split into multiple lines. Some help with this would be appreciated. We should also review rules that i've added and decide which should stay and which are missing

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    "templateStrings": true,
+    "unicodeCodePointEscapes": true,
+    "modules": true
+  },
+  "globals": {
+    "module": true,
+    // CommonJS module
+    "require": true,
+    // CommonJS require
+    "React": true,
```
Comment (mr-mig) [inline .eslintrc]: We do not have global React AFAIK

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    // ReactJs
+    "fetch": true,
```
Comment (mr-mig) [inline .eslintrc]: And there will be no fetch, this can be deleted

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    // Fetch js
+    /* MOCHA */
+    "describe": false,
+    "it": false,
```
Comment (mr-mig) [inline .eslintrc:105]: There should be `iit` and `ddescribe`, the globals for **forcing** single test and single spec

Comment (azakordonets) [inline .eslintrc:105]: I've added `iit` and `ddescribe` as a gulp task and included it into the `build` and `ci` tasks

File: gulpfile.js [hunk 4]
```diff
@@ -6,33 +6,34 @@ var path = require('path');
 var runSequence = require('run-sequence');
 var pkg = require('./package');
 var plugins = require('gulp-load-plugins')({
-    pattern: ['gulp-*', 'gulp.*', 'run\-sequence']
+  pattern: ['gulp-*', 'gulp.*', 'run\-sequence']
 });
+var ddescribeIit = require('gulp-acorn-ddescribe-iit');
```
Comment (mr-mig) [inline gulpfile.js:11]: What does thing thing do?

Comment (mr-mig) [inline gulpfile.js:11]: Ok, got it, great idea!

Comment (azakordonets) [inline gulpfile.js:11]: Sorry, didn't specify it in the pull request details. I took it from here - https://github.com/caitp/gulp-acorn-ddescribe-iit

**Pull request merged**